src/Fixer/PhpUnit/PhpUnitStaticAssertionCallsFixer.php

```diff
@@ -0,0 +1,351 @@
+        'greaterThanOrEqual' => true,
+        'identicalTo' => true,
+        'isEmpty' => true,
+        'isFalse' => true,
+        'isFinite' => true,
+        'isInfinite' => true,
+        'isInstanceOf' => true,
+        'isJson' => true,
+        'isNan' => true,
+        'isNull' => true,
+        'isReadable' => true,
+        'isTrue' => true,
+        'isType' => true,
+        'isWritable' => true,
+        'lessThan' => true,
+        'lessThanOrEqual' => true,
+        'logicalAnd' => true,
+        'logicalNot' => true,
+        'logicalOr' => true,
+        'logicalXor' => true,
+        'markTestIncomplete' => true,
+        'markTestSkipped' => true,
+        'matches' => true,
+        'matchesRegularExpression' => true,
+        'objectHasAttribute' => true,
+        'readAttribute' => true,
+        'resetCount' => true,
+        'stringContains' => true,
+        'stringEndsWith' => true,
+        'stringStartsWith' => true,
+
+        // TestCase methods
+        'any' => true,
+        'at' => true,
+        'atLeast' => true,
+        'atLeastOnce' => true,
+        'atMost' => true,
+        'exactly' => true,
+        'never' => true,
+        'onConsecutiveCalls' => true,
+        'once' => true,
+        'returnArgument' => true,
+        'returnCallback' => true,
+        'returnSelf' => true,
+        'returnValue' => true,
+        'returnValueMap' => true,
+        'throwException' => true,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'PHPUnit assertion calls must all be of the same type.',
+            [
+                new CodeSample(
+                    '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testMe()
+    {
+        $this->assertSame(1, 2);
+        self::assertSame(1, 2);
+        static::assertSame(1, 2);
+    }
+}
+'
+                ),
+            ],
+            null,
+            'Risky when PHPUnit methods are overridden or not accessible, or when project has PHPUnit incompatibilities.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_STRING]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach (array_reverse($this->findPhpUnitClasses($tokens)) as $indexes) {
+            $this->fixPhpUnitClass($tokens, $indexes[0], $indexes[1]);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('call_type', 'The call type to use for referring to PHPUnit methods.'))
```

Comment:
I believe we shall clone policy of `binary_operator_spaces` rule.
we have `default` strategy we want to apply, and then we have `operators` (here: `methods`) when we can override strategy.

eg, one may want to call all assertions with `->`, but `setUpBeforeClass` with `::`

Reply:
`binary_operator_spaces` is a style rule only.

This one is a SCA helper: if you want strict calls, you can't have exceptions.

I'm against your request, but of course I'll adopt it if you don't consider the above a valid point.

Reply:
I'm lookinig at the usage. like here, or in biggest framework - Symfony, assertions are called on `$this`, but `setUpBeforeClass` on class, not instance context, with `::`.
for that, it's not possible to use this rule without change of current "style", which is never easy to do. (well, easy to apply, hell to resolve conflicts on all PRs)

Reply:
> assertions are called on `$this`, but `setUpBeforeClass` on class, not instance context, with `::`.

What you describe here is a custom exception more than a custom style.

What about a list of excluded method, that will not be considered for fixing?
Something like:
```php
$config = [
    'php_unit_testcase_static_method_calls' => [
        'exclude_methods' => ['setUpBeforeClass', 'tearDownAfterClass']
    ],
];

Reply:
with that option, you will not be able to enforce `->assertTrue` but `::setUpBeforeClass`.
you will only enforce `->assertTrue` and leave `setUpBeforeClass` as is, even if it was called on `$this`.
with my proposal, you enforce style for every method

Reply:
Done